doc/src/when-to-use.md

```diff
@@ -0,0 +1,162 @@
+# When to use Symbolic Execution
+
+In the cryptocurrency world, it is exceedingly easy to lose a [lot of
+assets](https://chainsec.io/defi-hacks/) due to bugs. This risk can be alleviated
+via a good test approach. Testing is a [discipline of its
+own](https://en.wikipedia.org/wiki/Software_testing) that includes far more
+than simple test cases. Normally, it should include an understanding of what
+needs to be tested (i.e. scope) and for what types of issues (e.g.
+malicious/accidental). It normally consists of a test approach, documenting
+what has been tested and what remains to be tested. It should consist of a
+number of unit tests, module tests, integration tests, user acceptance tests,
+and exploratory testing. Written test cases normally consist of both positive
+tests (i.e. correct input, correct behaviour), and negative tests (i.e.
+incorrect input, no incorrect behaviour).
+
+## Fuzzing versus Symbolic Execution
+
+Fuzzing and symbolic execution fits within the [exploratory
```

Comment:
I wound't say that symbolic execution is a testing approach (also it is not claimed in the referenced wikipedia link).

Reply:
Ooops, thanks for catching this! I always thought that fuzzing is kind of an automation of exploratory testing, but that does sound bogus. I fixed it to:

```
Fuzzing tests usually have a set of (sometimes implicit) pre- and
postconditions, but the actual action (e.g. function call) is performed by an
external entity, the fuzzer. For C/C++ fuzzing, the implicit postcondition is
often e.g. that the system does not throw a segmentation fault. For EVM
bytecode, postconditions need to be explicit. Let's see an example:
```

What do you think?